pkg/controller/rest/verifiable/operation.go

```diff
@@ -164,14 +171,21 @@ func (o *Operation) GeneratePresentation(rw http.ResponseWriter, req *http.Reque
 //        200: presentationRes
 func (o *Operation) GeneratePresentationByID(rw http.ResponseWriter, req *http.Request) {
 	id := mux.Vars(req)["id"]
+	did := mux.Vars(req)["did"]
 
 	decodedID, err := base64.StdEncoding.DecodeString(id)
 	if err != nil {
 		rest.SendHTTPStatusError(rw, http.StatusBadRequest, verifiable.InvalidRequestErrorCode, err)
 		return
 	}
 
-	request := fmt.Sprintf(`{"id":"%s"}`, string(decodedID))
+	decodedDID, err := base64.StdEncoding.DecodeString(did)
```

Comment:
decode should happen inside command, no business logics/parsing in REST. Other in javascript also you may have to do it.